src/testhost.x86/DefaultEngineInvoker.cs

```diff
@@ -152,7 +163,14 @@ private Task StartProcessingAsync(ITestRequestHandler requestHandler, ITestHostM
                 () =>
                     {
                         // Wait for the connection to the sender and start processing requests from sender
-                if (requestHandler.WaitForRequestSenderConnection(ClientListenTimeOut))
+                        // Note that we are waiting here infinitely to connect to vstest.console, but at the same time vstest.console doesn't wait infinitely.
+                        // It has a default timeout of 60secs(which is configurable), & then it kills testhost.exe
+                        // The reason to wait infinitely, was remote debugging scenarios of UWP app,
+                        // in such cases after the app gets launched, VS debugger takes control of it, & causes a lot of delay, which frequently causes timeout with vstest.console.
+                        // One fix would be just double this timeout, but there is no telling how much time it can actually take.
+                        // Hence we are waiting here indefinelty, to avoid such guessed timeouts, & letting user kill the debugging if they feel it is taking too much time.
+                        // In other cases if vstest.console's timeout exceeds it will definitelty such down the app.
```

Comment:
Add EqtTrace Log to indicate starting/connected to vstest.console Communication.